model/config.go

```diff
@@ -335,55 +350,64 @@ func (i *istioConfigStore) RouteRulesBySource(instances []*ServiceInstance) []*p
 				continue
 			}
 		}
-		rules = append(rules, Config{Key: key, Content: rule})
+		rules = append(rules, config{Key: key, Spec: rule})
 	}
 	// sort by high precedence first, key string second (keys are unique)
 	sort.Slice(rules, func(i, j int) bool {
-		return rules[i].Content.(*proxyconfig.RouteRule).Precedence > rules[j].Content.(*proxyconfig.RouteRule).Precedence ||
-			(rules[i].Content.(*proxyconfig.RouteRule).Precedence == rules[j].Content.(*proxyconfig.RouteRule).Precedence &&
+		return rules[i].Spec.Precedence > rules[j].Spec.Precedence ||
+			(rules[i].Spec.Precedence == rules[j].Spec.Precedence &&
 				rules[i].Key < rules[j].Key)
 	})
 
 	// project to rules
 	out := make([]*proxyconfig.RouteRule, len(rules))
 	for i, rule := range rules {
-		out[i] = rule.Content.(*proxyconfig.RouteRule)
+		out[i] = rule.Spec
 	}
 	return out
 }
 
 func (i *istioConfigStore) IngressRules() map[string]*proxyconfig.IngressRule {
 	out := make(map[string]*proxyconfig.IngressRule)
-	rs, err := i.List(IngressRule.Type)
+	rs, err := i.List(IngressRule.Type, "")
 	if err != nil {
 		glog.V(2).Infof("IngressRules => %v", err)
 	}
 	for _, r := range rs {
-		if rule, ok := r.Content.(*proxyconfig.IngressRule); ok {
-			out[r.Key] = rule
+		if rule, ok := r.Spec.(*proxyconfig.IngressRule); ok {
+			out[r.Key()] = rule
 		}
 	}
 	return out
 }
 
 func (i *istioConfigStore) DestinationPolicies() []*proxyconfig.DestinationPolicy {
 	out := make([]*proxyconfig.DestinationPolicy, 0)
-	rs, err := i.List(DestinationPolicy.Type)
+	rs, err := i.List(DestinationPolicy.Type, "")
 	if err != nil {
 		glog.V(2).Infof("DestinationPolicies => %v", err)
 	}
 	for _, r := range rs {
-		if rule, ok := r.Content.(*proxyconfig.DestinationPolicy); ok {
+		if rule, ok := r.Spec.(*proxyconfig.DestinationPolicy); ok {
 			out = append(out, rule)
 		}
 	}
 	return out
 }
 
 func (i *istioConfigStore) DestinationPolicy(destination string, tags Tags) *proxyconfig.DestinationVersionPolicy {
-	value, exists, _ := i.Get(DestinationPolicy.Type, destination)
+	names := strings.Split(destination, ".")
+	name, namespace := "", ""
```

Comment:
we don't have a global namespace, CRDs are namespaces just like services themselves.